doc/source/whatsnew/v2.3.2.rst

```diff
@@ -25,7 +25,7 @@ Bug fixes
 - Fix :meth:`~DataFrame.to_json` with ``orient="table"`` to correctly use the
   "string" type in the JSON Table Schema for :class:`StringDtype` columns
   (:issue:`61889`)
-
+- Boolean operations (``|``, ``&``, ``^``) with bool-dtype objects on the left and :class:`StringDtype` objects on the right now cast the string to bool, with a deprecation warning (:issue:`60234`)
```

Comment:
Your PR might have fixed both, but I don't think this is specific to bool dtype being on the left and string on the right? Both orders of operation fails right now?

Reply:
The old behavior that we are trying to match (with object dtype) _does_ depend specifically on the ndarray[bool] being on the left

Reply:
Ah, I was only testing with string dtype to check that both were failing. You're right that for object this only worked in that order. Ignore me!